### PR TITLE
Added script for git push

### PR DESCRIPTION
**Pull request by andynygard**
We're in the process of migrating our maestro repositories to ci2 and have found that in order to publish to github pages we need to add push functionality (my understanding is that drone handles this instead via a plugin).

The intended use is as a two step process, e.g:
```bash
`... ci/sbt-ci-build-doc.sh && ci/ci-push-brach gh-pages`
```

I've kept configurability minimal (e.g. repo URL, keyfile are fixed), as I figure we can introduce that as the need arises.

This is based on [sbt-gh-pages-ssh](https://github.com/CommBank/ci/blob/master/sbt-gh-pages-ssh.sh) from ye olde ci.

### CODE REVIEW

File: src/main/bash/ci-push-branch.sh [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    echoerr "No branch specified. Exiting with error."
+    exit 1
+fi
+
+if [ ! -e $keyFile ]; then
+    echoerr "$keyFile does not exist. Exiting with error."
+    exit 1
+fi
+
+if [ $CI_BRANCH = "master" ] || [ ! -z $FORCE_PUBLISH ]; then
```
Comment (jsok) [inline src/main/bash/ci-push-branch.sh]: Is `Is_Release` more appropriate than the branch check?

**Comment by andynygard** [comment]
coveralls y u do dis

File: src/main/bash/ci-push-branch.sh (hunk 1 above)
Comment (andynygard) [inline src/main/bash/ci-push-branch.sh]: It looks like it, thanks for the suggestion.

I'm not sure if we're currently using `FORCE_PUBLISH` anywhere either, perhaps that could be removed to simplify things. @stephanh do you know if this parameter being used in zbi?

**Comment by andynygard** [comment]
OK I think changing `PATH` was giving coveralls a bad time, what with the ssh shading. Reverting `PATH` at the end of the script seems to have made things well again.

File: src/main/bash/ci-push-branch.sh [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+#
+#   Usage:
+#   ./ci-push-branch <branch>
+#
+#   Example:
+#   ./ci-push-branch gh-pages
+#
+#   Important:
+#   You need to place the deployment key at .ci/identity.pem in order to
+#   authenticate with git.
```
Comment (rowandavies) [inline src/main/bash/ci-push-branch.sh]: Just to check: this is essentially the same as what was in  `.travis/deploy-key.pem` in `CommBank/ci`?

**Comment by rowandavies** [comment]
Update the README to add this script to the list, maybe with a hint like `e.g., for gh-pages`.

File: src/main/bash/ci-push-branch.sh (hunk 2 above)
Comment (rowandavies) [inline src/main/bash/ci-push-branch.sh]: Also, is it worth having instructions like those in step 6 of:  https://github.com/CommBank/ci/blob/a53f524f613711a0e91063396639e9c96d0530ee/README.md#travis-setup

Comment (andynygard) [inline src/main/bash/ci-push-branch.sh]: Yep, I just wanted to be agnostic of travis. I suppose we could put it at `.ci/deploy-key.pem`, as it's unlikely we'd use any other kind of key?

Comment (andynygard) [inline src/main/bash/ci-push-branch.sh]: I've updated this to be `deploy-key.pem`.

**Comment by andynygard** [comment]
Good point! I've updated this in the latest push, incl. some instructions on how to set up for travis users.

File: README.md [hunk 3]
```diff
@@ -34,11 +34,12 @@ If you don't use a versioned approach for pulling in these scripts, there is no
...
 * ``ci-nonrelease.bsh`` - Perform an action, only if the Branch/PR is a NON-Release Branch
 * ``ci-release.bsh`` - Perform the action, only if the Branch/PR is a Release Branch
-* ``ci-publish-site.bsh`` - Perform the 
+* ``ci-publish-site.bsh`` - commit a site's directory as the sole content of a given branch
+* ``ci-push-branch.bsh`` - push a branch to the github repo
 * ``dockermake.bsh`` - Build all ``Dockerfile`` images in the Repo 
 * ``docker-support.bsh`` - ``setup`` and ``publish`` docker images (implies a build step between each)
 * ``gh-commit-comment.bsh`` - uses the GitHub API to create a comment against a specific git commit-sha.
-* ``sbt-ci-build-doc.bsh`` - uses sbt to build documentation for the project
+* ``sbt-ci-build-doc.bsh`` - uses sbt to build documentation for the project and commit to ``gh-pages``
```
Comment (vineethvarghese) [inline README.md]: Should we say `commit to ``gh-pages`` branch`?

Comment (andynygard) [inline README.md]: Yep, updated

File: src/main/bash/ci-push-branch.sh (hunk 1 above)
Comment (rowandavies) [inline src/main/bash/ci-push-branch.sh]: I think we should keep it.

File: README.md [hunk 4]
```diff
@@ -74,4 +75,21 @@ these scripts.
...
+   file as a template.
+2. Install the travis client. Instructions are available at https://github.com/travis-ci/travis.rb. **Please be aware that the installation and usage of the travis client often result in network and SSL Errors from within the office network (including wifi). Tethering to  a 4G device is one workaround for this.**
+3. Login. `travis login --pro` or `travis login --org`.
+4. From project folder (same directory as `.travis.yml`), enable your project with Travis CI by running the command:
+   `travis enable --pro` or `travis enable --org`.
+5. Add the encrypted artifactory username and password by running these commands in the same directory as `.travis.yml`:
+   - `travis encrypt ARTIFACTORY_USERNAME=... --add env.global`
+   - `travis encrypt ARTIFACTORY_PASSWORD=... --add env.global`
+6. [Optional to publish documentation] **For public repos only** Create a branch called `gh-pages` for the project and push it to github. Then add the private key for omnia-bamboo as an encrypted file.
+   1. Get the private key
```
Comment (rowandavies) [inline README.md]: I guess we don’t have an official process for this, I think in practice this is done by asking around.  We could add `somehow` just to make it clear that the instructions don’t say how to do this.

File: README.md [hunk 5]
```diff
@@ -74,4 +75,21 @@ these scripts.
   and the CI scripts append the time stamp and commish at the beginning of the build.
   e.g. ``(file) VERSION: 2.0.3`` --> ``ci-2.0.3-20160517230553-92fd78d``
 
-
+## Usage
+
+### Travis Setup
```
Comment (rowandavies) [inline README.md]: Maybe make it clear that Travis is not the only option - perhaps `### For Travis`

**Comment by rowandavies** [comment]
LGTM, just two tiny things to consider for the README.  👍

**Comment by vineethvarghese** [comment]
:+1:

**Pull request merged**